### PR TITLE
Fix skip logic and validation in Social Support Questionnaire

### PR DESCRIPTION
**Pull request by vishnuravi**
# Fixes Social Support Questionnaire

## :recycle: Current situation & Problem
The *Social Support Questionnaire* which is provided as an example survey in the Spezi Template Application has several issues:

- The survey ends prematurely after the question regarding the user's preferred contact method because the enable-when expressions are not written correctly.
- Field validation fails due to an incorrect extension URI in multiple fields and an invalid regular expression for validating a US phone number.

## :bulb: Proposed solution
- Fixes the enable-when expressions to show the appropriate follow-up question after the preferred contact method is selected.
- Uses the correct extension URI which is recognized by `ResearchKitOnFHIR` and fixes the regular expression.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: TemplateApplication/Resources/SocialSupportQuestionnaire.json [hunk 1]
```diff
@@ -360,6 +293,95 @@
...
+      ]
+    },
+    {
+      "linkId": "86290b0a-017e-4193-8707-dc0c2146f0eb",
+      "type": "string",
+      "text": "What is your e-mail?",
+      "extension": [
+        {
+          "url": "http://hl7.org/fhir/StructureDefinition/regex",
+          "valueString": "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"
```
Comment (Supereg) [inline TemplateApplication/Resources/SocialSupportQuestionnaire.json:356]: Do we have guidelines on how we want to handle E-Mail validation and verification?

We currently have a mixed set of E-Mail regular expression around all Spezi modules (e.g. SpeziAccount maintains one as well). All regex I encountered so far don't specify their source and also match against a grammar that is diverging from [RF5322](https://datatracker.ietf.org/doc/html/rfc5322#section-3.4.1). While the above regex seems fine in that sense that it isn't too restrictive (it still captures less that what is allowed with the RFC afaik), it still wanted to use this opportunity to start this discussion.

There are generally two concerns:
* We cause adoption of email Regular Expressions within the framework ecosystem and dependent projects that may reject perfectly valid E-Mail addresses.
* Having a RFC compliant or "properly" formatted email address doesn't give us any information about its existence. Ultimately, an email may only considered valid if the user can proof being able to receive emails on that address.

So there are really two questions to this discussion:
* What type of Regex do we agree on to use within the Spezi ecosystem (maybe even use-case-dependent?). As an example, in the upcoming SpeziAccount release I'm currently striving for a more minimal validation as it is backed by E-Mail validation of the Account Service anyways (see [here](https://github.com/StanfordSpezi/SpeziAccount/blob/297fc05e57ed568ca2a7a3b0c284e1e02c6f05cb/Sources/SpeziAccount/Model/Validation/ValidationRule%2BDefaults.swift#L48)). There are great StackOverflow discussions providing more insights about [parsing](https://stackoverflow.com/a/48170419) and the [general problem domain](https://stackoverflow.com/a/201378).
* Validating E-Mails (and phone numbers) is a problem other people using the framework might likewise encounter. Do we have the ability to rely on sending E-Mail verifications for data provided in questionnaires? Could we provide a way to source that data from other components (e.g. the currently logged in user from SpeziAccount)?

Might make sense to move this discussion somewhere else. This should not block your PR.

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the update @vishnuravi, great that this can be fixed by just replacing the example questionnaire! Once the comment by @Supereg is addressed you can merge the PR from my side 🚀

File: TemplateApplication/Resources/SocialSupportQuestionnaire.json (hunk 1 above)
Comment (vishnuravi) [reply TemplateApplication/Resources/SocialSupportQuestionnaire.json:356]: Hi @PSchmiedmayer, it is [defined in Phoenix]( https://github.com/StanfordBDHG/phoenix/blob/4bfefd80fcacd0e4f46666708f5eb0ba98d69e24/src/components/Question/ValidationAnswerTypes/ValidationAnswerTypeString.tsx#L25). We can change it. There is no specific regex defined in the FHIR standard to my knowledge.

Comment (PSchmiedmayer) [reply TemplateApplication/Resources/SocialSupportQuestionnaire.json:356]: Sounds great, let's change that to the very simple regex that @Supereg is using [here]( https://github.com/StanfordSpezi/SpeziAccount/blob/297fc05e57ed568ca2a7a3b0c284e1e02c6f05cb/Sources/SpeziAccount/Model/Validation/ValidationRule%2BDefaults.swift#L48).

Different discussion, but I think we should generally also think about the other RegEx expressions in Phoenix, e.g. the Zip code is US-specific (maybe just renaming it to US-Zip is the way to go) and the URL Regex doesn't seem to match, e.g. new top level domains that are longer than 6 characters are not matched and the focus on `www` and `ftp` might be too specific for the URL. But that might be a discussion over in Phoenix 🚀

Let's just update the RegEx here and merge the PR 🥳

**Comment by vishnuravi** [comment]
@PSchmiedmayer I've incorporated your and @Supereg's feedback regarding the email regex. All tests have passed, however, there seems to be a configuration issue with CodeQL. Please take a look when you get a chance. Thanks.

**Review by PSchmiedmayer — APPROVED** [review]

**Pull request merged**